demos/coreMQTT_Agent/mqtt_agent_task.c

```diff
@@ -0,0 +1,1025 @@
+ */
+static NetworkContext_t xNetworkContext;
+
+/**
+ * @brief The parameters for the network context using a TLS channel.
+ */
+static SecureSocketsTransportParams_t secureSocketsTransportParams;
+
+/**
+ * @brief Global entry time into the application to use as a reference timestamp
+ * in the #prvGetTimeMs function. #prvGetTimeMs will always return the difference
+ * between the current time and the global entry time. This will reduce the chances
+ * of overflow for the 32 bit unsigned integer used for holding the timestamp.
+ */
+static uint32_t ulGlobalEntryTimeMs;
+
+MQTTAgentContext_t xGlobalMqttAgentContext;
+
+static uint8_t xNetworkBuffer[ MQTT_AGENT_NETWORK_BUFFER_SIZE ];
+
+static MQTTAgentMessageContext_t xCommandQueue;
+
+static struct DemoParams taskParameters[ democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE ];
+
+/**
+ * @brief The global array of subscription elements.
+ *
+ * @note No thread safety is required to this array, since the updates the array
+ * elements are done only from one task at a time. The subscription manager
+ * implementation expects that the array of the subscription elements used for
+ * storing subscriptions to be initialized to 0. As this is a global array, it
+ * will be initialized to 0 by default.
+ */
+SubscriptionElement_t xGlobalSubscriptionList[ SUBSCRIPTION_MANAGER_MAX_SUBSCRIPTIONS ];
+
+/*-----------------------------------------------------------*/
+
+/*
+ * @brief Create the task that demonstrates the MQTT Connection sharing demo.
+ */
+int RunCoreMqttAgentDemo( bool awsIotMqttMode,
+                          const char * pIdentifier,
+                          void * pNetworkServerInfo,
+                          void * pNetworkCredentialInfo,
+                          const void * pNetworkInterface )
+{
+    BaseType_t xNetworkStatus = pdFAIL;
+    BaseType_t xResult = pdFALSE;
+    BaseType_t xNetworkConnectionCreated = pdFALSE;
+    uint32_t ulNotification = 0UL;
+
+    uint32_t ulDemoCount = 0UL;
+    uint32_t ulDemoSuccessCount = 0UL;
+    int ret = EXIT_SUCCESS;
+
+    ( void ) awsIotMqttMode;
+    ( void ) pIdentifier;
+    ( void ) pNetworkServerInfo;
+    ( void ) pNetworkCredentialInfo;
+    ( void ) pNetworkInterface;
+
+    for( ulDemoCount = 0UL; ( ulDemoCount < democonfigMQTT_MAX_DEMO_COUNT ); ulDemoCount++ )
+    {
+        ret = prvConnectAndCreateDemoTasks( NULL );
+
+        if( ret == EXIT_SUCCESS )
+        {
+            LogInfo( ( "Demo iteration %lu successful.", ulDemoCount ) );
+            break;
+        }
+        else if( ulDemoCount < ( democonfigMQTT_MAX_DEMO_COUNT - 1 ) )
+        {
+            LogWarn( ( "Demo iteration %lu failed. Retrying...", ulDemoCount ) );
+        }
+        else
+        {
+            LogError( ( "All %d iterations failed", democonfigMQTT_MAX_DEMO_COUNT ) );
+        }
+    }
+
+    return ret;
+}
+
+/*-----------------------------------------------------------*/
+
+static MQTTStatus_t prvMQTTInit( void )
```

Comment:
(Nit) Maybe, this is more fitting of a name?
```suggestion
static MQTTStatus_t prvMQTTAgentInit( void )
```

Reply:
Sure